### PR TITLE
Add normalization function for DuplicateKeyTracker

### LINKED ISSUE #355: Expand `jobState.findEntity()` interface to match case-insensitive IDs
Some integrations return case-insensitive IDs that need to be mapped back to entities that exist in the job state. The SDK should make the `findEntity` methods more flexible to meet these needs.

Possible implementations:
```typescript
// with options argument
findEntity: (_key: string, options?: { matchCaseInsensitive?: boolean }) => Promise<Entity | null>;
// with regex
findEntity: (_key: string | RegExp) => Promise<Entity | null>;
// future... with regex or callback?
findEntity: ((_key: string | RegExp) => Promise<Entity | null>) | ((callback: (_key: string) => boolean) => Promise<Entity | null>;
```

### PR DESCRIPTION
**Pull request by ndowmon**
The azure integration has been hitting some issues where entity `id`s are returned with inconsistent casing. This makes the `findEntity` function useless in most cases, and prohibits us from identifying valid relationships across an environment. This PR allows users to pass a `keyNormalizationFunction` into an `InvocationConfig` that transforms keys before storing them (or retrieving them from) the `DuplicateKeyTracker`.

Closes #355

### CODE REVIEW

**Review by austinkelleher — APPROVED** [review]
Nice. This seems like a good approach. 👍

**Review by aiwilliams — CHANGES_REQUESTED** [review]

File: packages/integration-sdk-core/src/types/config.ts [hunk 1]
```diff
@@ -8,13 +8,29 @@ import {
...
+  /**
+   * Normalization transform for tracking keys in an integration. Allows
+   * entities to be tracked and looked up using the provided transformation.
+   * One use case of this normalization function is to track duplicates
+   * among case-insensitive keys.
+   *
+   * Example:
+   *   (_key: string) => _key.toLowerCase()
+   */
+  keyNormalizationFunction?: KeyNormalizationFunction;
```
[the remark this answers is not in the document]
Comment (ndowmon) [reply packages/integration-sdk-core/src/types/config.ts:23]: That works. I can change the parameter name but leave the type as `type KeyNormalizationFunction` (I don't think `type NormalizeGraphObjectKey` is a good description for a type, what do you think?)

**Comment by ndowmon** [comment]
Also merged https://github.com/JupiterOne/sdk/pull/359 into this PR.

File: packages/integration-sdk-core/src/types/config.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (aiwilliams) [reply packages/integration-sdk-core/src/types/config.ts]: I think `KeyNormalizationFunction` is perfect! Thanks for the variable rename.

**Review by aiwilliams — APPROVED** [review]

**Pull request merged**